src/Microsoft.Health.Fhir.Core/Features/Search/SearchOptionsFactory.cs

```diff
@@ -151,6 +161,29 @@ public SearchOptions Create(string resourceType, IReadOnlyList<Tuple<string, str
 
             options.UnsupportedSearchParams = unsupportedSearchParameters;
 
+            options.CompartmentType = null;
+
+            // Initialize enum to undefined value
+            var parsedCompartmentType = (CompartmentType)(-1);
```

Comment:
you could use nullable here

Reply:
actually you dont really need to declare this since you could just do `Enum.TryParse(compartmentType, out CompartmentType parsedCompartmentType)` below